common/configuration/asm.go

```diff
@@ -0,0 +1,162 @@
+package configuration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/TBD54566975/ftl/internal/slices"
+
+	. "github.com/alecthomas/types/optional" //nolint:stylecheck
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager/types"
+	"github.com/aws/smithy-go"
+)
+
+// ASM implements Resolver and Provider for AWS Secrets Manager (ASM).
+//
+// The resolver does a direct/proxy map from a Ref to a URL, module.name <-> asm://module.name and does not access ASM at all.
+type ASM struct {
+	client secretsmanager.Client
```

Comment:
This field will have to public in order for the type to be constructible by external callers.